src/Symfony/Component/Translation/Bridge/Lokalise/LokaliseProvider.php

```diff
@@ -176,6 +188,113 @@ private function exportFiles(array $locales, array $domains): array
         return array_combine($reformattedLanguages, $responseContent['files']);
     }
 
+    /**
+     * @see https://developers.lokalise.com/reference/download-files-async
+     */
+    private function exportFilesAsync(array $locales, array $domains): array
+    {
+        $response = $this->client->request('POST', 'files/async-download', [
+            'json' => [
+                'format' => 'symfony_xliff',
+                'original_filenames' => true,
+                'filter_langs' => array_values($locales),
+                'filter_filenames' => array_map($this->getLokaliseFilenameFromDomain(...), $domains),
+                'export_empty_as' => 'skip',
+                'replace_breaks' => false,
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new ProviderException(\sprintf('Unable to export translations from Lokalise: "%s".', $response->getContent(false)), $response);
+        }
+
+        $processId = $response->toArray()['process_id'];
+        $attempt = 0;
+        while (true) {
+            $response = $this->client->request('GET', \sprintf('processes/%s', $processId));
+            $process = $response->toArray()['process'];
+            if (\in_array($process['status'], self::FAILED_PROCESS_STATUS, true)) {
+                throw new ProviderException(\sprintf('Unable to export translations from Lokalise: "%s".', $response->getContent(false)), $response);
+            }
+            if (self::SUCESS_PROCESS_STATUS === $process['status']) {
+                $downloadUrl = $process['details']['download_url'];
+                break;
+            }
+            ++$attempt;
+            usleep(500000 * $attempt);
+        }
+
+        $response = $this->client->request('GET', $downloadUrl, ['buffer' => true]);
```

Comment:
no need to buffer since we write by chunk:
```suggestion
        $response = $this->client->request('GET', $downloadUrl, ['buffer' => false]);
```

(the alternative would be to pass a file handler instead of a boolean, then wait for the last chunk below, but it's also fine as you wrote it)